packages/drivers/debugger/src/fluidDebuggerController.ts

```diff
@@ -148,14 +149,18 @@ export class DebugReplayController extends ReplayController implements IDebugger
         reader.readAsText(file, "utf-8");
     }
 
-    public async onDownloadOpsButtonClick(): Promise<string> {
+    public async onDownloadOpsButtonClick(anonymize: boolean): Promise<string> {
         if (this.documentService === undefined) {
             throw new Error("DocumentService required");
         }
 
         const documentDeltaStorageService = await this.documentService.connectToDeltaStorage();
         const messages = await this.fetchOpsFromDeltaStorage(documentDeltaStorageService);
-        return JSON.stringify(messages, undefined, 2);
+
+        const sanitizer = new Sanitizer(messages, false /* fullScrub */, true /* noBail */);
```

Comment:
Are we potentially exposing PII with noBail = false?
It feels like if we fail here, we should stop.

Reply:
I'm not even sure it should be an argument. I.e. code either makes some educated guess on risk and moves on, or fails. But asking that question to users feels wrong - I feel like all users would just want to say - if you can't anonymize, do not send my data, or at least let me know before you send.

Reply:
Makes sense, will change to fail on any errors.  The tool doesn't actually send anything automatically, and should (later) allow the user to preview the scrubbed document, at which point I think it could e reverted.  My main concern was that we need some sort of knowledge about the json structure in order to anonymize in a way that allows ops to be replayed/the document to be reloaded, and if we always fail whenever we encounter some unexpected structure we're stuck playing catch up to fix those as they come up, and be stuck with a broken feature in the meantime.

Reply:
Also I'm pretty sure we will always expose some PII if we want the sanitized document to be replayable.  When adding the anonymized preview, we will need to expose some of these options to the user and what they mean so that they're informed about what's happening instead of making those decisions and assumptions under the hood for them.